src/marqo/tensor_search/utils.py

```diff
@@ -309,4 +308,10 @@ def wrap(*args, **kw):
         time_taken = timer() - t0
         r[key] = round(time_taken * 1000)
         return r
-    return wrap
\ No newline at end of file
+    return wrap
+
+
+def generate_batches(seq: Sequence, batch_size):
+    """Yields batches of length k from the sequence."""
```

Comment:
this function will raise an error when `batch_size=0`. is this expected or we set batch_size = 1 instead?

By the way, do we have any validation when reading env variables in marqo?

Reply:
Added validation for when batch_size is 0 (it doesn't really make sense for it to be 0).

I've also added a function to validate env var max batch size